### PR TITLE
Refactor error modals

### PR DESCRIPTION
**Pull request by ds300**
https://artsyproduct.atlassian.net/browse/PURCHASE-792 

Finishing up the ErrorModal -> ModalDialog refactor.

Can't wait to clean up those tests 💆‍♂️

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 45ab0bd9b99e6ef4c823c681fb8297f421a3a05a

https://deploy-preview-1875--artsy-reaction.netlify.com

File: src/Apps/Order/Routes/Counter/index.tsx [hunk 1]
```diff
@@ -102,13 +97,11 @@ export class CounterRoute extends Component<CounterProps, CounterState> {
     })
   }
 
-  onMutationError(errors, errorModalTitle?, errorModalMessage?) {
+  onMutationError(errors, title?, message?) {
     logger.error(errors)
+    this.props.dialog.showErrorDialog({ message, title })
```
Comment (damassi) [inline src/Apps/Order/Routes/Counter/index.tsx]: 👍

File: src/Apps/Order/Routes/Review/index.tsx [hunk 2]
```diff
@@ -240,19 +233,14 @@ export class ReviewRoute extends Component<ReviewProps, ReviewState> {
...
-    errorModalTitle?,
-    errorModalMessage?,
-    errorModalCtaAction?
-  ) {
+  onMutationError(error, title?, message?, onContinue?) {
     logger.error(error)
+    this.props.dialog
+      .showErrorDialog({ message, title })
+      // tslint:disable-next-line:no-empty
+      .then(onContinue || (() => {}))
```
Comment (damassi) [inline src/Apps/Order/Routes/Review/index.tsx]: Ah, now I understand this promise api 😄

**Comment by sweir27** [comment]
This PR looks 🎉 ! One question since I'm coming at these conversations late-- it looks like the `Dialog` component embeds some assumption that it'll be an `error` dialog. Anticipating the eventual desire to move these to [palette](https://github.com/artsy/palette/issues/82), do you feel like we should make any changes now? Or it's in a good place to be eventually refactored?

Apologies if you've already thought about this/commented elsewhere!

**Comment by ds300** [comment]
@sweir27  The `ModalDialog` is good to move to palette as-is. That doesn't have any error-specific code.

Not sure it makes sense to move the state/context management stuff (in `Dialogs.tsx`) to palette at this point. I expect it to be a generally useful pattern, but it feels premature to decide it's the one true way. So I'd prefer to wait and see what requirements emerge in other parts of artsy.net.

File: src/Apps/Order/Routes/__tests__/Accept.test.tsx [hunk 3]
```diff
@@ -207,18 +209,25 @@ describe("Accept seller offer", () => {
...
+      const errorComponent = component.find(ModalDialog)
       expect(errorComponent.props().show).toBe(true)
       expect(errorComponent.text()).toContain("An error occurred")
       expect(errorComponent.text()).toContain(
         "Something went wrong. Please try again or contact orders@artsy.net."
       )
 
       component.find(ModalButton).simulate("click")
-      expect(component.find(ErrorModal).props().show).toBe(false)
+
```
Comment (pepopowitz) [inline src/Apps/Order/Routes/__tests__/Accept.test.tsx]: These bits (lines 223-227) are repeated throughout the tests - we 1. render the component, 2. verify the modal shows up, 3. click the button to dismiss the modal, then 4. verify that the modal is gone. Is there value in repeating steps 3&4 in almost every test, or would it make more sense to write a single test against `injectDialog` that verifies this instead?

**Comment by damassi** [comment]
> state/context management
I agree that this seems a bit premature, might be good just to get the component portion in which we can then populate with data

File: src/Apps/Order/Routes/__tests__/Accept.test.tsx (hunk 3 above)
Comment (ds300) [reply src/Apps/Order/Routes/__tests__/Accept.test.tsx]: Yeah that would be good. Shall we do it as a follow up or would you prefer to have it taken care of in this PR?

Comment (pepopowitz) [reply src/Apps/Order/Routes/__tests__/Accept.test.tsx]: Follow-up would be great!

**Comment by pepopowitz** [comment]
This looks great!!!! Anything remaining with comments above can be handled as a follow-up.

**Pull request merged**